TrafficSimulator/Assets/Old UI/GraphChartFeed.cs

```diff
@@ -5,50 +5,58 @@ namespace Old_UI
 {
     public class GraphChartFeed : MonoBehaviour
     {
+        private enum TimeSpan
+        {
+            None,
+            AllTime,
+            ThreeMinutes,
+            ThirtySeconds
+        }
+
         private GraphChartBase _graph;
         private WorldDataGatherer _worldDataGatherer;
+        private TimeSpan _currentGraph = TimeSpan.None; 
 
         private void Start()
         {
             _worldDataGatherer = GameObject.Find("RoadSystem").GetComponent<WorldDataGatherer>();
             _graph = GetComponent<GraphChartBase>();
 
             if (_graph != null)
-            {
                 _graph.Scrollable = false;
-                LoadAllTimeGraph();
-            }
         }
 
         public void LoadAllTimeGraph()
         {
-            LoadGraph(0, _worldDataGatherer.TotalSecondsElapsed);
+            LoadGraph(TimeSpan.AllTime, _worldDataGatherer.TotalSecondsElapsed);
         }
 
         public void LoadLast3MinGraph()
         {
-            int totalSecondsElapsed = _worldDataGatherer.TotalSecondsElapsed;
-            int startIndex = Mathf.Max(totalSecondsElapsed - 180, 0);
-            LoadGraph(startIndex, 180);
+            LoadGraph(TimeSpan.ThreeMinutes, 180);
         }
 
         public void LoadLast30SecondsGraph()
         {
-            int totalSecondsElapsed = _worldDataGatherer.TotalSecondsElapsed;
-            int startIndex = Mathf.Max(totalSecondsElapsed - 30, 0);
-            LoadGraph(startIndex, 30);
+            LoadGraph(TimeSpan.ThirtySeconds, 30);
         }
 
-        private void LoadGraph(int startIndex, int length)
+        private void LoadGraph(TimeSpan timeSpan, int timeSpanInSeconds)
```

Comment:
Just add the seconds to the TimeSpan enum, and int cast them here. Then you don't need the `timeSpanInSeconds`

```
private enum TimeSpan
        {
            None = 0,
            AllTime = -1,
            ThreeMinutes = 180,
            ThirtySeconds = 30
        }
```

Then just a check for if timespan is AllTime, then use the total elapsed time instead